File: array/ReverseString/ReverseString.java
package array.ReverseString;


public class ReverseString {
    
    public static String reverseString(String str){

        String result = "";
        final int length = str.length();

        //adding the chars on the new string starting in the end
        for (int i = length - 1; i >= 0; i--){
            result+= str.charAt(i);
        }
        return result;
    }

    //Recommended way of doing it
    //Its gonna to to return a new string and reverse it
    public static String reverseStringBetter(String str){
        return new StringBuilder(str).reverse().toString();
    }

    public static void main (String[] args){

        String name = "Gabriela Melnik Carvalho";
        System.out.println(reverseString(name));
        System.out.println(reverseStringBetter(name));
    }
}

File: array/ReverseString/reverseString.js
//Create a function that reverses a string


function reverseString(str) {
  //check input
  if(!str || str.length < 2 || typeof str !== 'string'){
    return 'There is something wrong, this is invalid!';
  }
  const arrStr = [];
  const length = str.length;

  //fill array with string
  for (let i = 0; i < length; i++) {
    arrStr[i] = str.charAt(i);
  }
  //reverse array
  arrStr.reverse();

  return arrStr.join('');
}

//Transform the string in an array, reverse it and transform it again into a String
function reverseStringEasier(str) {
    if (!str || str.length < 2 || typeof str !== 'string') {
      return 'There is something wrong, this is invalid!';
    }
  return str.split('').reverse().join('');
}

const reverseStringFancier = str => [...str].reverse().join('');

const name = 'Gabriela Melnik Carvalho';
console.log(reverseString(name));
console.log(reverseStringEasier(name));
console.log(reverseStringFancier(name));